src/swap/components/SwapButton.tsx

```diff
@@ -3,14 +3,11 @@ import { Spinner } from '../../internal/loading/Spinner';
 import { useSwapContext } from './SwapProvider';
 import type { SwapButtonReact } from '../types';
 
-export function SwapButton({
-  className,
-  disabled = false,
-  onSubmit,
```

Comment:
So what's going on to `onSubmit` going forwar?

Reply:
removed and no longer overridable

Reply:
if we remove that, we still need a way to listen to `onErrors` and `onSuccess`.

Somewhere we need to give back to the developers the information of the Swap experience.

Reply:
added hooks for both of these